File: core/src/ru/myitschool/dcrawler/math/NegativeAction.java
package ru.myitschool.dcrawler.math;

public class NegativeAction extends MathAction {

    MathAction action;

    public NegativeAction(MathAction action) {
        this.action = action;
    }

    @Override
    public int act() {
        return -action.act();
    }

    /** returns negotiated max of inner action*/
    @Override
    public int max() {
        return -action.max();
    }

    /** returns negotiated min of inner action*/
    @Override
    public int min() {
        return -action.min();
    }

    @Override
    public String getDescription() {
        return "-(" + action.getDescription() + ")";
    }
}

File: core/src/com/wolg_vlad/dcrawler/math/NegativeAction.kt
package com.wolg_vlad.dcrawler.math

class NegativeAction(var action: MathAction) : MathAction() {
    override fun act(): Int {
        return -action.act()
    }

    /** returns negotiated max of inner action */
    override fun max(): Int {
        return -action.max()
    }

    /** returns negotiated min of inner action */
    override fun min(): Int {
        return -action.min()
    }

    override val description: String?
        get() = "-(" + action.description + ")"

}